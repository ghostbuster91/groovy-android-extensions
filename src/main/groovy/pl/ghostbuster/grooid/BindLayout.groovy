package pl.ghostbuster.grooid

import org.codehaus.groovy.transform.GroovyASTTransformationClass

import java.lang.annotation.ElementType
import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy
import java.lang.annotation.Target

@Retention(RetentionPolicy.SOURCE)
@Target([ElementType.TYPE])
@GroovyASTTransformationClass(classes = [BindLayoutASTTransformation])
@interface BindLayout {
}
