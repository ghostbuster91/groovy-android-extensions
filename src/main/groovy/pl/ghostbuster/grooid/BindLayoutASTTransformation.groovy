package pl.ghostbuster.grooid

import org.codehaus.groovy.ast.ASTNode
import org.codehaus.groovy.control.CompilePhase
import org.codehaus.groovy.control.SourceUnit
import org.codehaus.groovy.transform.AbstractASTTransformation
import org.codehaus.groovy.transform.GroovyASTTransformation

@GroovyASTTransformation(phase = CompilePhase.SEMANTIC_ANALYSIS)
class BindLayoutASTTransformation extends AbstractASTTransformation {

    @Override
    void visit(ASTNode[] nodes, SourceUnit source) {

    }

}
