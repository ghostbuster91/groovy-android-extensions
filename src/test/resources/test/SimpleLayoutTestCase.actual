package test

import pl.ghostbuster.grooid.*


@BindLayout({R.layout.simple_layout})
class SimpleLayoutTestCase {


}
