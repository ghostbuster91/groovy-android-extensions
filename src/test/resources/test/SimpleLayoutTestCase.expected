package test

import android.widget.LinearLayout
import android.view.View
import pl.ghostbuster.grooid.*

class SimpleLayoutTestCase {

    public final LinearLayout someLinearLayoutId


    public SimpleLayoutTestCase(View itemView){
          someLinearLayoutId = (LinearLayout) itemView.findViewById(R.id.some_linear_layout_id)
    }
}
